sycl/unittests/context_device/HasExtensionWordBoundary.cpp

```diff
@@ -0,0 +1,147 @@
+//==---- HasExtensionWordBoundary.cpp --- Test word boundary fix ----------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This test verifies that has_extension correctly matches full extension names
+// and doesn't match partial substrings.
+//
+//===----------------------------------------------------------------------===//
+
+#include <detail/device_impl.hpp>
+#include <gtest/gtest.h>
+#include <helpers/UrMock.hpp>
+#include <sycl/sycl.hpp>
+#include <ur_mock_helpers.hpp>
+
+using namespace sycl;
+
+static std::string MockExtensions = "";
+
+static ur_result_t redefinedDeviceGetInfo(void *pParams) {
+  auto params = *static_cast<ur_device_get_info_params_t *>(pParams);
+
+  if (*params.ppropName == UR_DEVICE_INFO_EXTENSIONS) {
+    // override extensions query with mock data
+    if (*params.ppPropValue) {
+      size_t len = MockExtensions.length() + 1;
+      if (*params.ppropSize >= len)
+        std::memcpy(*params.ppPropValue, MockExtensions.c_str(), len);
+    }
+    if (*params.ppPropSizeRet)
+      **params.ppPropSizeRet = MockExtensions.length() + 1;
+
+    return UR_RESULT_SUCCESS;
+  }
+
+  // delegate to the default mock
+  return sycl::unittest::MockAdapter::mock_urDeviceGetInfo(pParams);
```

Comment:
Alternatively, you can use `set_before_callback` and return some error code here (e.g. `UR_RESULT_ERROR_UNSUPPORTED_FEATURE`) which basically does the same thing.